images/upload-gitlab-failure-logs/upload_gitlab_failure_logs.py

```diff
@@ -120,6 +123,40 @@ def assign_error_taxonomy(job_input_data: dict):
     return
 
 
+def collect_pod_status(job_input_data: dict[str, Any], job_trace: str):
+    """Collect k8s info about this job and store it in the OpenSearch record"""
+    # Scan job logs to infer the name of the pod this job was executed on
```

Comment:
Can we check for the presence of "Using Kubernetes executor" in the job_trace here, and return early if it's not found?

Example of job from cloud runner (has this string):
https://gitlab.spack.io/spack/spack/-/jobs/6623467

Example of job from on-premises runner (doesn't have this string):
https://gitlab.spack.io/spack/spack/-/jobs/6621535